src/hotspot/os/linux/os_linux.cpp

```diff
@@ -2989,7 +2988,7 @@ int os::Linux::sched_getcpu_syscall(void) {
   retval = vgetcpu(&cpu, nullptr, nullptr);
 #endif
 
-  return (retval == -1) ? retval : cpu;
+  return (retval == -1) ? checked_cast<int>(retval) : cpu;
```

Comment:
If retval is -1 you can just return -1. And if cpu is uint and the function returns int then surely we need a cast on cpu here?

Reply:
I don't have -Wsign-conversion on.  I'm not that much of a masochist.  Fixed return value though.

Reply:
So `-Wconversion` is mainly of use in detecting overflow/overflow when dealing with variables of different size, yet unsigned to signed conversion can result in the exact the same problem, but that only gets detected by a separate `-Wsign-conversion` flag? \<blegghh>